frontend/public/components/catalog.jsx

```diff
@@ -0,0 +1,220 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import * as fuzzy from 'fuzzysearch';
+import * as PropTypes from 'prop-types';
+import { Helmet } from 'react-helmet';
+
+import { FLAGS, connectToFlags, flagPending } from '../features';
+import { Firehose, NavTitle, StatusBox } from './utils';
+import { CatalogList } from './catalog-items';
+import { getBuilderSpecTagName, getImageStreamTagAnnotation } from './image-stream';
+import { serviceClassDisplayName } from '../module/k8s';
+import { getCSCIcon, getCSCImage } from './cluster-service-class-icon';
+
+class CatalogListPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleFilterChange = this.handleFilterChange.bind(this);
+    this.clearFilter = this.clearFilter.bind(this);
+
+    this.state = {
+      filterValue: '',
+      items: [],
+      filteredItems: [],
+    };
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const {clusterserviceclasses, namespace } = this.props;
+    const {filterValue} = this.state;
+    if (namespace !== prevProps.namespace || clusterserviceclasses !== prevProps.clusterserviceclasses) {
+      this.createCatalogData();
+    } else if (filterValue !== prevState.filterValue) {
+      const filteredItems = this.filterItems(this.state.items);
+      this.setState({
+        filteredItems
+      });
+    }
+  }
+
+  filterItems(items) {
+    const {filterValue} = this.state;
+
+    if (!filterValue) {
+      return items;
+    }
+
+    const filterString = filterValue.toLowerCase();
+    return _.filter(items, item => {
+      return fuzzy(filterString, _.get(item, 'metadata.name', ''));
+    });
+  }
+
+  createCatalogData() {
+    const {clusterserviceclasses, imagestreams, loaded} = this.props;
+
+    if (!loaded) {
+      return;
+    }
+
+    const clusterServiceClassItems = this.normalizeClusterServiceClasses(clusterserviceclasses.data, 'ClusterServiceClass');
+    const imagestreamsItems = this.normalizeImagestreams(imagestreams.data, 'ImageStream');
+
+    const items = [
+      ...clusterServiceClassItems,
+      ...imagestreamsItems
+    ];
+
+    const filteredItems = this.filterItems(items);
+    this.setState({items, filteredItems});
+  }
+
+  normalizeClusterServiceClasses(serviceClasses, kind) {
+    const filteredServiceClasses = _.filter(serviceClasses, serviceClass => {
+      return !serviceClass.status.removedFromBrokerCatalog;
+    });
+
+    return _.map(filteredServiceClasses, serviceClass => {
+      const tileName = serviceClassDisplayName(serviceClass);
+      const iconClass = getCSCIcon(serviceClass);
+      const imgUrl = getCSCImage(serviceClass, iconClass);
```

Comment:
Prefer names `getServiceClassIcon` and `getServiceClassImage`. Eventually we will need to handle namespaced service classes, not just cluster service classes.

Reply:
> Do the tooltips pop up immediately when you mouse over items? It looks to me like that could be really distracting.

Tooltips only appear when you hover over the description text, not the top part of the tile.

Reply:
OK, but the description is still most of the tile... Does the tooltip have a delay?

Reply:
The tooltip for the description doesn't show unless the description text is truncated.

Reply:
Ok, I'll readdress after #534 is merged